src/kernel.rs

```diff
@@ -29,10 +35,28 @@ impl<'a, K> Kernel<'a, K> {
     ///
     /// # Panics
     ///
-    /// If the `x` or `y` is outside of the width or height of the kernel.
+    /// If the `y * kernel.width + x` is outside of the kernel data.
     #[inline]
-    pub fn at(&self, x: u32, y: u32) -> &K {
-        &self.data[(y * self.width + x) as usize]
+    pub fn get(&self, x: u32, y: u32) -> K {
```

Comment:
Why did you rename `at()` to `get()`? This was discussed as not being a good idea as `get()` normally returns an option.